daemon/errors.go

```diff
@@ -132,35 +132,50 @@ func (e startInvalidConfigError) Error() string {
 
 func (e startInvalidConfigError) InvalidParameter() {} // Is this right???
 
-func translateContainerdStartErr(cmd string, setExitCode func(int), err error) error {
+// exitStatus is the exit-code as returned by translateContainerdStartErr
+type exitStatus = int
+
+const (
+	exitNoError     exitStatus = 0   // no error occurred (err is nil)
+	exitUnknown     exitStatus = -1  // unknown error (callers should not update the container's exitcode)
+	exitEaccess     exitStatus = 126 // container cmd can't be invoked (permission denied)
+	exitCmdNotFound exitStatus = 127 // container cmd not found/does not exist or invalid bind-mount
+)
+
+// translateContainerdStartErr converts the error returned by containerd (if any)
+// when starting a container, and returns the corresponding exitStatus for the
+// container and an errdefs error (either errdefs.ErrInvalidParameter or errdefs.ErrUnknown).
+func translateContainerdStartErr(err error) (exitCode exitStatus, exitErr error) {
+	if err == nil {
+		return exitNoError, nil
```

Comment:
Hmm shouldn't this case _also_ leave the container's exit status alone?  (Sorry for not being more familiar with this code :see_no_evil:)

Looking at the previous implementation, it seems to not do anything very different in the `nil` case - maybe just because the function wasn't called in the `nil` case?  (The whole translation feels odd IMO, but I get it.)

Reply:
>  maybe just because the function wasn't called in the nil case?

Yes, so previously the container state was modified within this function, and the function wasn't called if `err`  was nil

IN this PR the update happens if `err != nil`, so basically "any value" would be ok here;

```go
if exitCode, err := translateContainerdStartErr(err); err != nil {
	container.SetExitCode(exitCode)
	return err
}
```

(So if `err` is nil to begin with, it would not update the exit code)

Reply:
Ahhh got it, thanks for clarifying :+1: